core/src/main/java/com/netflix/conductor/core/dal/ExecutionDAOFacade.java

```diff
@@ -509,6 +555,29 @@ public void removeTask(String taskId) {
         executionDAO.removeTask(taskId);
     }
 
+    private void removeTaskIndex(WorkflowModel workflow, TaskModel task, boolean archiveTask)
+            throws JsonProcessingException {
+        if (archiveTask) {
+            if (task.getStatus().isTerminal()) {
+                // Only allow archival if task is in terminal state
+                // DO NOT archive async, since if archival errors out, task data will be lost
+                indexDAO.updateTask(
```

Comment:
The archival process for task does not need a rawJSON field because the archival process for workflow, contains all of the tasks within in the stringified JSON blob stored in its rawJSON field.

Reply:
Removed! Will keep the tasks just for auditing purposes.